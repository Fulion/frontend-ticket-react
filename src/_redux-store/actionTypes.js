export const actionTypes = {
  GET_USER_REQUEST: "GET_USER_REQUEST",
  GET_USER_SUCCESS: "GET_USER_SUCCESS",
  GET_USER_FAILED: "GET_USER_FAILED",

  LOGIN_REQUEST: "LOGIN_REQUEST",
  LOGIN_SUCCESS: "LOGIN_SUCCESS",
  LOGIN_FAILED: "LOGIN_FAILED",

  SIGNUP_REQUEST: "SIGNUP_REQUEST",
  SIGNUP_SUCCESS: "SIGNUP_SUCCESS",
  SIGNUP_FAILED: "SIGNUP_FAILED",

  LOGOUT_REQUEST: "LOGOUT_REQUEST",
  LOGOUT_SUCCESS: "LOGOUT_SUCCESS",
  LOGOUT_FAILED: "LOGOUT_FAILED",

  // Ticket 
  
  CREATE_TICKET_REQUEST: "CREATE_TICKET_REQUEST",
  CREATE_TICKET_SUCCESS: "CREATE_TICKET_SUCCESS",
  CREATE_TICKET_FAILED: "CREATE_TICKET_FAILED",

  GET_TICKETS_REQUEST: "GET_TICKETS_REQUEST",
  GET_TICKETS_SUCCESS: "GET_TICKETS_SUCCESS",
  GET_TICKETS_FAILED: "GET_TICKETS_FAILED",

  GET_TICKET_BY_ID_REQUEST: "GET_TICKET_BY_ID_REQUEST",
  GET_TICKET_BY_ID_SUCCESS: "GET_TICKET_BY_ID_SUCCESS",
  GET_TICKET_BY_ID_FAILED: "GET_TICKET_BY_ID_FAILED",

  UPDATE_TICKET_REQUEST: "UPDATE_TICKET_REQUEST",
  UPDATE_TICKET_SUCCESS: "UPDATE_TICKET_SUCCESS",
  UPDATE_TICKET_FAILED: "UPDATE_TICKET_FAILED",

  CLOSED_TICKET_REQUEST: "CLOSED_TICKET_REQUEST",
  CLOSED_TICKET_SUCCESS: "CLOSED_TICKET_SUCCESS",
  CLOSED_TICKET_FAILED: "CLOSED_TICKET_FAILED",

  DELETE_TICKET_REQUEST: "DELETE_TICKET_REQUEST",
  DELETE_TICKET_SUCCESS: "DELETE_TICKET_SUCCESS",
  DELETE_TICKET_FAILED: "DELETE_TICKET_FAILED",

  UPDATE_PROPERTY: "UPDATE_PROPERTY",
};
