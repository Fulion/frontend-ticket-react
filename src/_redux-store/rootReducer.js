import { combineReducers } from "redux";
import { createBrowserHistory } from "history";
import { persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";

// import all reducer
import { appReducer, ticketReducer } from "./reducers";

const persistConfigs = {
  key: "sw-ticket-store",
  storage,
  blacklist: [],
};

// now we generate an application history object.
export const routeHistory = createBrowserHistory();

// then we create the app reducer, by combining all other reducers
const rootReducer = combineReducers({
  root: appReducer,
  ticket: ticketReducer,
});

// finally we export the persisted reducer
export const persistedRootReducer = persistReducer(persistConfigs, rootReducer);
