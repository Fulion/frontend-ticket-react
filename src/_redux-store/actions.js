import { actionTypes } from "./actionTypes";

export const authActions = {
  // Login
  loginRequest: (email, password) => {
    return {
      type: actionTypes.LOGIN_REQUEST,
      payload: { email, password },
    };
  },
  loginRequestSuccess: (authData) => {
    return {
      type: actionTypes.LOGIN_SUCCESS,
      payload: { authData },
    };
  },
  loginRequestFailed: (error) => {
    return {
      type: actionTypes.LOGIN_FAILED,
      payload: { error },
    };
  },
  // SignUp
  signUpRequest: (user) => {
    return {
      type: actionTypes.SIGNUP_REQUEST,
      payload: { user },
    };
  },
  signUpRequestSuccess: () => {
    return {
      type: actionTypes.SIGNUP_SUCCESS,
      payload: {},
    };
  },
  signUpRequestFailed: (error) => {
    return {
      type: actionTypes.SIGNUP_FAILED,
      payload: { error },
    };
  },
  // Logout
  logoutRequest: (token) => {
    return {
      type: actionTypes.LOGOUT_REQUEST,
      payload: { token },
    };
  },
  logoutRequestSuccess: () => {
    return {
      type: actionTypes.LOGOUT_SUCCESS,
      payload: {},
    };
  },
  logoutRequestFailed: (error) => {
    return {
      type: actionTypes.LOGOUT_FAILED,
      payload: { error },
    };
  },
  // user
  getUserRequest: (token) => {
    return {
      type: actionTypes.GET_USER_REQUEST,
      payload: { token },
    };
  },
  getUserRequestSuccess: (userData) => {
    return {
      type: actionTypes.GET_USER_SUCCESS,
      payload: { userData },
    };
  },
  getUserRequestFailed: (error) => {
    return {
      type: actionTypes.GET_USER_FAILED,
      payload: { error },
    };
  },
};

export const ticketActions = {
  // create
  createTicket: (ticket, token) => {
    return {
      type: actionTypes.CREATE_TICKET_REQUEST,
      payload: { ticket, token },
    };
  },
  createTicketSuccess: (response) => {
    return {
      type: actionTypes.CREATE_TICKET_SUCCESS,
      payload: { response },
    };
  },
  createTicketFailed: (error) => {
    return {
      type: actionTypes.CREATE_TICKET_FAILED,
      payload: { error },
    };
  },
  // update
  updateTicket: (ticket, ticketId, token) => {
    return {
      type: actionTypes.UPDATE_TICKET_REQUEST,
      payload: { ticket, ticketId, token },
    };
  },
  updateTicketSuccess: (ticket) => {
    return {
      type: actionTypes.UPDATE_TICKET_SUCCESS,
      payload: { ticket },
    };
  },
  updateTicketFailed: (error) => {
    return {
      type: actionTypes.UPDATE_TICKET_FAILED,
      payload: { error },
    };
  },

  //get tickets

  getTickets: (token) => {
    return {
      type: actionTypes.GET_TICKETS_REQUEST,
      payload: { token },
    };
  },
  getTicketsSuccess: (tickets) => {
    return {
      type: actionTypes.GET_TICKETS_SUCCESS,
      payload: { tickets },
    };
  },
  getTicketsFailed: (error) => {
    return {
      type: actionTypes.GET_TICKETS_FAILED,
      payload: { error },
    };
  },

  //get ticket by id

  getTicketById: (ticketId, token) => {
    return {
      type: actionTypes.GET_TICKET_BY_ID_REQUEST,
      payload: { ticketId, token },
    };
  },
  getTicketByIdSuccess: (ticket) => {
    return {
      type: actionTypes.GET_TICKET_BY_ID_SUCCESS,
      payload: { ticket },
    };
  },

  getTicketByIdFailed: (error) => {
    return {
      type: actionTypes.GET_TICKET_BY_ID_FAILED,
      payload: { error },
    };
  },

  //update status closed

  closedTicket: (ticketId, token) => {
    return {
      type: actionTypes.CLOSED_TICKET_REQUEST,
      payload: { ticketId, token },
    };
  },
  closedTicketSuccess: (ticket) => {
    return {
      type: actionTypes.CLOSED_TICKET_SUCCESS,
      payload: { ticket },
    };
  },
  closedTicketFailed: (error) => {
    return {
      type: actionTypes.CLOSED_TICKET_FAILED,
      payload: { error },
    };
  },

  //delete Tickets

  deleteTicket: (ticketId, token) => {
    return {
      type: actionTypes.DELETE_TICKET_REQUEST,
      payload: { ticketId, token },
    };
  },
  deleteTicketSuccess: () => {
    return {
      type: actionTypes.DELETE_TICKET_SUCCESS,
    };
  },
  deleteTicketFailed: (error) => {
    return {
      type: actionTypes.DELETE_TICKET_FAILED,
      payload: { error },
    };
  },

  updatePropertyHandler: (property) => {
    return {
      type: actionTypes.UPDATE_PROPERTY,
      payload: { property },
    };
  },
};
