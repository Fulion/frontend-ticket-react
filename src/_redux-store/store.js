import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";
import { persistStore } from "redux-persist";

import { persistedRootReducer } from "./rootReducer";
import { rootSaga } from "./rootSaga";

const sagaMiddleware = createSagaMiddleware();

//* production da bu kisim duzenlenecek.
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const middleware = composeEnhancers(applyMiddleware(sagaMiddleware));

const store = createStore(persistedRootReducer, middleware);

const persistor = persistStore(store);

export { store, persistor };

sagaMiddleware.run(rootSaga);
