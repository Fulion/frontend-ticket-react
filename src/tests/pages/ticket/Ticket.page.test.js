import {render, screen} from '@testing-library/react';
import Ticket from './../../../pages/ticket/Ticket.page';
import {Provider} from "react-redux";
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import {Router} from "react-router-dom";
import {createMemoryHistory} from "history";
import userEvent from "@testing-library/user-event";
import {ticketActions} from "../../../_redux-store/actions";

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => ({ state: { email: 'someemail@something.com' } }),
    // useHistory: () => ({ push: jest.fn()}),
}));

describe('Ticket tests', () => {


    test('ticket', () => {


        const middlewares = [thunk];

        const mockStore = configureMockStore(middlewares);

        const store = mockStore(
            {
                tid: "test",
                ticket: {
                    ticket: "someticket",
                    isLoading: false,
                    deleting: null,
                },
                root: {
                    token: "token",
                }
            }
        );
        const tree = render(
            <Provider store={store}>
                <Ticket />);
            </Provider>
        )

        const linkElement = screen.getByText(/SchlieÃŸen Sie die Tickets ein/i);
        expect(linkElement).toBeInTheDocument();
        expect(tree).toMatchSnapshot();
    });

    test('ticket_is_deleted in history', () => {

        const middlewares = [thunk];

        const mockStore = configureMockStore(middlewares);

        const store = mockStore(
            {
                tid: "test",
                ticket: {
                    ticket: "someticket",
                    isLoading: false,
                    deleting: "ticket_is_deleted",
                },
                root: {
                    token: "token",
                }
            }
        );
        const history = createMemoryHistory()
        const pushSyp = jest.spyOn(history, 'push');

        const tree = render(
            <Provider store={store}>
                <Router history={history}>
                    <Ticket />
                </Router>
            </Provider>
        )
        expect(pushSyp).toHaveBeenCalledWith("/dashboard");
        expect(tree).toMatchSnapshot();

    });

    test('handleOnSubmit', () => {

        const middlewares = [thunk];

        const mockStore = configureMockStore(middlewares);

        const store = mockStore(
            {
                tid: "test",
                ticket: {
                    ticket: "someticket",
                    isLoading: false,
                    deleting: null,
                },
                root: {
                    token: "token",
                }
            }
        );

        // console.log(aktualisierenTicket);

        const tree = render(
            <Provider store={store}>
                <Ticket/>
            </Provider>
        )

        const aktualisierenTicketButton = screen.getAllByText(/Reply/i)[1];
        // const handleOnSubmit = container.getAllByRole("handleOnSubmit");
        // console.log(aktualisierenTicket);
        // const handleOnChange = jest.spyOn(aktualisierenTicket, 'handleOnChange');

        userEvent.click(aktualisierenTicketButton);
        expect(tree).toMatchSnapshot();

        // expect(ticketActions.updateTicket).toHaveBeenCalled();
        // expect(handleOnSubmit).toHaveBeenCalled();
        // expect(handleOnChange).not.toHaveBeenCalled();
        // expect(pushSyp).toHaveBeenCalled();
    });

    test('handleOnChange', () => {

        const middlewares = [thunk];

        const mockStore = configureMockStore(middlewares);

        const store = mockStore(
            {
                tid: "test",
                ticket: {
                    ticket: "someticket",
                    isLoading: false,
                    deleting: null,
                },
                root: {
                    token: "token",
                }
            }
        );

        // console.log(aktualisierenTicket);

        const tree = render(
            <Provider store={store}>
                <Ticket/>
            </Provider>
        )

        const textbox = screen.getByRole("textbox");
        // const handleOnSubmit = container.getAllByRole("handleOnSubmit");
        console.log(textbox);
        // const handleOnChange = jest.spyOn(aktualisierenTicket, 'handleOnChange');

        userEvent.paste(textbox, "some value");
        expect(tree).toMatchSnapshot();

        // expect(handleOnSubmit).toHaveBeenCalled();
        // expect(handleOnChange).not.toHaveBeenCalled();
        // expect(pushSyp).toHaveBeenCalled();
    });
})
