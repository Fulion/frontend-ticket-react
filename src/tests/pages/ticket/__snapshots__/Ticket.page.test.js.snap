// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket tests first test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <nav
                aria-label="breadcrumb"
              >
                <ol
                  class="breadcrumb"
                >
                  <li
                    class="breadcrumb-item"
                  >
                    <a
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    aria-current="page"
                    class="breadcrumb-item active"
                  >
                    Ticket
                  </li>
                </ol>
              </nav>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="text-weight-bolder text-secondary col"
            >
              <div
                class="subject"
              >
                Subject :
              </div>
              <div
                class="date"
              >
                Das Ticket würde geöffnet :
              </div>
              <div
                class="status"
              >
                Status :
              </div>
            </div>
            <div
              class="text-right col"
            >
              <button
                class="btn btn-dark"
                type="button"
              >
                Schließen Sie die Tickets ein
              </button>
              <button
                class="btn btn-danger"
                type="button"
              >
                Delete
              </button>
            </div>
          </div>
          <div
            class="mt-4 row"
          >
            <div
              class="col"
            />
          </div>
          <hr />
          <div
            class="mt-4 row"
          >
            <div
              class="col"
            >
              <form
                class=""
              >
                <label
                  class="form-label"
                >
                  Reply
                </label>
                <small
                  class="form-text"
                />
                <textarea
                  class="form-control"
                  name="detail"
                  row="5"
                />
                <div
                  class="text-right mt-3 mb-3"
                >
                  <button
                    class="btn btn-danger"
                    type="submit"
                  >
                    Reply
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      );
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <nav
              aria-label="breadcrumb"
            >
              <ol
                class="breadcrumb"
              >
                <li
                  class="breadcrumb-item"
                >
                  <a
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  aria-current="page"
                  class="breadcrumb-item active"
                >
                  Ticket
                </li>
              </ol>
            </nav>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="text-weight-bolder text-secondary col"
          >
            <div
              class="subject"
            >
              Subject :
            </div>
            <div
              class="date"
            >
              Das Ticket würde geöffnet :
            </div>
            <div
              class="status"
            >
              Status :
            </div>
          </div>
          <div
            class="text-right col"
          >
            <button
              class="btn btn-dark"
              type="button"
            >
              Schließen Sie die Tickets ein
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          class="mt-4 row"
        >
          <div
            class="col"
          />
        </div>
        <hr />
        <div
          class="mt-4 row"
        >
          <div
            class="col"
          >
            <form
              class=""
            >
              <label
                class="form-label"
              >
                Reply
              </label>
              <small
                class="form-text"
              />
              <textarea
                class="form-control"
                name="detail"
                row="5"
              />
              <div
                class="text-right mt-3 mb-3"
              >
                <button
                  class="btn btn-danger"
                  type="submit"
                >
                  Reply
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
    );
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Ticket tests handleOnChange 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <nav
                aria-label="breadcrumb"
              >
                <ol
                  class="breadcrumb"
                >
                  <li
                    class="breadcrumb-item"
                  >
                    <a
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    aria-current="page"
                    class="breadcrumb-item active"
                  >
                    Ticket
                  </li>
                </ol>
              </nav>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="text-weight-bolder text-secondary col"
            >
              <div
                class="subject"
              >
                Subject :
              </div>
              <div
                class="date"
              >
                Das Ticket würde geöffnet :
              </div>
              <div
                class="status"
              >
                Status :
              </div>
            </div>
            <div
              class="text-right col"
            >
              <button
                class="btn btn-dark"
                type="button"
              >
                Schließen Sie die Tickets ein
              </button>
              <button
                class="btn btn-danger"
                type="button"
              >
                Delete
              </button>
            </div>
          </div>
          <div
            class="mt-4 row"
          >
            <div
              class="col"
            />
          </div>
          <hr />
          <div
            class="mt-4 row"
          >
            <div
              class="col"
            >
              <form
                class=""
              >
                <label
                  class="form-label"
                >
                  Reply
                </label>
                <small
                  class="form-text"
                />
                <textarea
                  class="form-control"
                  name="detail"
                  row="5"
                >
                  some value
                </textarea>
                <div
                  class="text-right mt-3 mb-3"
                >
                  <button
                    class="btn btn-danger"
                    type="submit"
                  >
                    Reply
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <nav
              aria-label="breadcrumb"
            >
              <ol
                class="breadcrumb"
              >
                <li
                  class="breadcrumb-item"
                >
                  <a
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  aria-current="page"
                  class="breadcrumb-item active"
                >
                  Ticket
                </li>
              </ol>
            </nav>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="text-weight-bolder text-secondary col"
          >
            <div
              class="subject"
            >
              Subject :
            </div>
            <div
              class="date"
            >
              Das Ticket würde geöffnet :
            </div>
            <div
              class="status"
            >
              Status :
            </div>
          </div>
          <div
            class="text-right col"
          >
            <button
              class="btn btn-dark"
              type="button"
            >
              Schließen Sie die Tickets ein
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          class="mt-4 row"
        >
          <div
            class="col"
          />
        </div>
        <hr />
        <div
          class="mt-4 row"
        >
          <div
            class="col"
          >
            <form
              class=""
            >
              <label
                class="form-label"
              >
                Reply
              </label>
              <small
                class="form-text"
              />
              <textarea
                class="form-control"
                name="detail"
                row="5"
              >
                some value
              </textarea>
              <div
                class="text-right mt-3 mb-3"
              >
                <button
                  class="btn btn-danger"
                  type="submit"
                >
                  Reply
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Ticket tests handleOnSubmit 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <nav
                aria-label="breadcrumb"
              >
                <ol
                  class="breadcrumb"
                >
                  <li
                    class="breadcrumb-item"
                  >
                    <a
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    aria-current="page"
                    class="breadcrumb-item active"
                  >
                    Ticket
                  </li>
                </ol>
              </nav>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="text-weight-bolder text-secondary col"
            >
              <div
                class="subject"
              >
                Subject :
              </div>
              <div
                class="date"
              >
                Das Ticket würde geöffnet :
              </div>
              <div
                class="status"
              >
                Status :
              </div>
            </div>
            <div
              class="text-right col"
            >
              <button
                class="btn btn-dark"
                type="button"
              >
                Schließen Sie die Tickets ein
              </button>
              <button
                class="btn btn-danger"
                type="button"
              >
                Delete
              </button>
            </div>
          </div>
          <div
            class="mt-4 row"
          >
            <div
              class="col"
            />
          </div>
          <hr />
          <div
            class="mt-4 row"
          >
            <div
              class="col"
            >
              <form
                class=""
              >
                <label
                  class="form-label"
                >
                  Reply
                </label>
                <small
                  class="form-text"
                />
                <textarea
                  class="form-control"
                  name="detail"
                  row="5"
                />
                <div
                  class="text-right mt-3 mb-3"
                >
                  <button
                    class="btn btn-danger"
                    type="submit"
                  >
                    Reply
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <nav
              aria-label="breadcrumb"
            >
              <ol
                class="breadcrumb"
              >
                <li
                  class="breadcrumb-item"
                >
                  <a
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  aria-current="page"
                  class="breadcrumb-item active"
                >
                  Ticket
                </li>
              </ol>
            </nav>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="text-weight-bolder text-secondary col"
          >
            <div
              class="subject"
            >
              Subject :
            </div>
            <div
              class="date"
            >
              Das Ticket würde geöffnet :
            </div>
            <div
              class="status"
            >
              Status :
            </div>
          </div>
          <div
            class="text-right col"
          >
            <button
              class="btn btn-dark"
              type="button"
            >
              Schließen Sie die Tickets ein
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          class="mt-4 row"
        >
          <div
            class="col"
          />
        </div>
        <hr />
        <div
          class="mt-4 row"
        >
          <div
            class="col"
          >
            <form
              class=""
            >
              <label
                class="form-label"
              >
                Reply
              </label>
              <small
                class="form-text"
              />
              <textarea
                class="form-control"
                name="detail"
                row="5"
              />
              <div
                class="text-right mt-3 mb-3"
              >
                <button
                  class="btn btn-danger"
                  type="submit"
                >
                  Reply
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Ticket tests ticket_is_deleted in history 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <nav
                aria-label="breadcrumb"
              >
                <ol
                  class="breadcrumb"
                >
                  <li
                    class="breadcrumb-item"
                  >
                    <a
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    aria-current="page"
                    class="breadcrumb-item active"
                  >
                    Ticket
                  </li>
                </ol>
              </nav>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="text-weight-bolder text-secondary col"
            >
              <div
                class="subject"
              >
                Subject :
              </div>
              <div
                class="date"
              >
                Das Ticket würde geöffnet :
              </div>
              <div
                class="status"
              >
                Status :
              </div>
            </div>
            <div
              class="text-right col"
            >
              <button
                class="btn btn-dark"
                type="button"
              >
                Schließen Sie die Tickets ein
              </button>
              <button
                class="btn btn-danger"
                type="button"
              >
                Delete
              </button>
            </div>
          </div>
          <div
            class="mt-4 row"
          >
            <div
              class="col"
            />
          </div>
          <hr />
          <div
            class="mt-4 row"
          >
            <div
              class="col"
            >
              <form
                class=""
              >
                <label
                  class="form-label"
                >
                  Reply
                </label>
                <small
                  class="form-text"
                />
                <textarea
                  class="form-control"
                  name="detail"
                  row="5"
                />
                <div
                  class="text-right mt-3 mb-3"
                >
                  <button
                    class="btn btn-danger"
                    type="submit"
                  >
                    Reply
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <nav
              aria-label="breadcrumb"
            >
              <ol
                class="breadcrumb"
              >
                <li
                  class="breadcrumb-item"
                >
                  <a
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  aria-current="page"
                  class="breadcrumb-item active"
                >
                  Ticket
                </li>
              </ol>
            </nav>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="text-weight-bolder text-secondary col"
          >
            <div
              class="subject"
            >
              Subject :
            </div>
            <div
              class="date"
            >
              Das Ticket würde geöffnet :
            </div>
            <div
              class="status"
            >
              Status :
            </div>
          </div>
          <div
            class="text-right col"
          >
            <button
              class="btn btn-dark"
              type="button"
            >
              Schließen Sie die Tickets ein
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          class="mt-4 row"
        >
          <div
            class="col"
          />
        </div>
        <hr />
        <div
          class="mt-4 row"
        >
          <div
            class="col"
          >
            <form
              class=""
            >
              <label
                class="form-label"
              >
                Reply
              </label>
              <small
                class="form-text"
              />
              <textarea
                class="form-control"
                name="detail"
                row="5"
              />
              <div
                class="text-right mt-3 mb-3"
              >
                <button
                  class="btn btn-danger"
                  type="submit"
                >
                  Reply
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
